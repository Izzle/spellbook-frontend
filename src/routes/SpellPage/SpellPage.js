import React, { Component } from 'react';
import SpellContext from '../../contexts/SpellContext';
import SpellApiService from '../../services/spell-api-service';
// most of this code is scratch
// its not really being used, it was stuff i was typing before I knew what I was doing
// feel free to erase everything in here
export default class SpellPage extends Component {
  static defaultProps = {
      match: { params: { id: '1'} }
  };

  static contextType = SpellContext;

  componentDidMount() {
    const { id } = this.props.match.params;
    
    console.log(id);
    SpellApiService.getSpellById(id)
      .then(this.context.setSpells)
      .catch(this.context.setError);
  }

  render() {
    console.log(this.props.match);
    return (
      <div className='SpellPage'>
        <h2>An page for individual spells!</h2>
      </div>
    )
  }
}