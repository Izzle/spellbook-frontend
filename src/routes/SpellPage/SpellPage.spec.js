import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import SpellPage from './SpellPage';
import { BrowserRouter } from 'react-router-dom';


describe('<SpellPage />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><SpellPage /></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<BrowserRouter><SpellPage /></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});