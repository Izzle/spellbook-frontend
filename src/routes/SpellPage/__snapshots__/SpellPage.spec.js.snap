// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpellPage /> renders this UI as expected 1`] = `
<div
  className="SpellPage"
>
  <h2>
    An page for individual spell!
  </h2>
  <p />
  <div
    className="Spell__main"
    onClick={[Function]}
  >
    <h3 />
    <p>
      Level: undefined | undefined | undefined
    </p>
    <p>
      undefined | undefined | Components: undefined
    </p>
    <p>
      Description: undefined
    </p>
  </div>
</div>
`;
